### ISSUE TITLE
Unable to compile (Origin: bugzilla #725633)

### ISSUE THREAD

**Issue by doxygen** [doxygen/doxygen#5434]
status RESOLVED severity *major* in component *build* for *---*
Reported in version *1.8.6* on platform *Other*
Assigned to: Dimitri van Heesch

On 2014-03-04 04:11:19 +0000, Naram Qashat wrote:

> Hi, I'm the port maintainer for the FreeBSD version of doxygen, and I'm running into an error upon trying to compile 1.8.6 so I can update the port. Upon getting to where it should compile config.cpp, I get the following:
> 
> clang++ -c -O2 -pipe -fno-strict-aliasing -pipe -fno-exceptions -Wall -W -I../qtools -I/usr/local/include -o ../objects/config.o config.cpp
> config.l:1692:12: error: use of undeclared identifier 'configYY_flex_debug'
>   printlex(yy_flex_debug, TRUE, __FILE__, fn);
>            ^
> config.cpp:6:23: note: expanded from macro 'yy_flex_debug'
> #define yy_flex_debug configYY_flex_debug
>                       ^
> config.l:1694:12: error: use of undeclared identifier 'configYY_flex_debug'
>   printlex(yy_flex_debug, FALSE, __FILE__, fn);
>            ^
> config.cpp:6:23: note: expanded from macro 'yy_flex_debug'
> #define yy_flex_debug configYY_flex_debug
>                       ^
> 2 errors generated.
> 
> I get the same error (but only once) when I use g++ instead. While the FreeBSD port does have some edits done to it to allow it to compile there, none of those edits affect config.l at all. The only thing I see right before the compile is the use of /usr/bin/flex on config.l, but I can see nothing offhand that would cause the compile error.
> 
> As soon as I can get past this error, I can see if there are any other issues with the FreeBSD compile. I know in the past I've had issues with getting doxygen to compile it's PDF docs, regardless of if I use the old teTeX that FreeBSD had been using or the newer TeXLive that was put into the ports tree recently.

On 2014-03-04 17:34:01 +0000, albert wrote:

> Looks like you are using a too old version of flex / bison. I'm using version 2.5.35 and 2.3.37.
> Please check your flex / bison version.

On 2014-03-05 00:50:36 +0000, Naram Qashat wrote:

> That did the trick. The flex that comes with FreeBSD at /usr/bin/flex was only version 2.5.4, the one from ports at /usr/local/bin/flex is version 2.5.38.
> 
> I'm still running into an error compiling the PDF docs. It's the same issue I've been having with bug # 700858 basically. I have longtable at /usr/local/share/texmf-dist/tex/latex/tools/longtable.sty and have even tried to add \usepackage{longtable} right after \RequirePackage{longtable}, but that hasn't helped either. I've been at a loss as to why I keep having this problem. I had a different issue with generating the PDF docs with 1.8.3.1 (which is what the FreeBSD port is currently at), but since 1.8.4 is when I've had the longtable issue.
> 
> (If it would be better to continue discussion of this issue on the other bug, then let's do so.)

On 2014-03-05 10:06:09 +0000, albert wrote:

> I've set the bug to resolved fixed (as there is no possibility to mention it is a local installation issue).
> 
> Regarding the bug # 700858 this should not be discussed with this bug as it is a separate issue.
